### ISSUE TITLE
Add --fix option to automatically resolve lint errors/warnings

### ISSUE THREAD

**Issue by elodszopos** [sasstools/sass-lint#904]
Hey folk, not necessarily an issue, more of a feature request / question. 

I'm sure you're aware of the [fix feature of eslint](http://eslint.org/docs/user-guide/command-line-interface#fix).

Would this be something that we could expect to see for sass-lint too? It'd be absolutely amazing. Tried looking through issues but haven't found anything similar. 

Maybe if there's an already on-going effort or if there were discussions about this I could provide some assistance?

Thank you for the awesome tool and looking forward to hearing back!

**Comment by DanPurdy** [comment]
HI @elodszopos this was actually being worked on [here](https://github.com/sasstools/sass-lint/pull/778) but unfortunately the author wasn't keen on following our suggestions on how to get this work released / following our contribution guidelines which made it really hard for us to manage or ultimately merge.

We'd obviously be very happy to see this effort come back to life. We'd request that PR's were broken up on a rule by rule basis and included the relevant tests and documentation but I'd be happy to help with these areas if someone wants to just write the functionality.

We even have this branch https://github.com/sasstools/sass-lint/tree/feature/auto-fix ready for all PR's to be applied to so we could release a bulk of these fix issues and test them independently of our main develop branch.

TLDR: We'd love to see this functionality! 🎉

**Comment by elodszopos** [comment]
I can see that there was a little back-and-forth with this one. Guidelines are important. Everyone's time matters. I'll most definitely take a look.

So, CLI integration in one PR, followed by others that are on a per rule basis with tests? Make sense @DanPurdy ? Thanks for the quick info and the detailed answer, really helpful.

**Comment by ryancoughlin** [comment]
Bump to get this reviewed and merged? This would be perfect.

**Comment by wlingke** [comment]
+1

I hate having to manually fix extra whitespaces or nextline issues..

**Comment by heyimnowi** [comment]
+1

**Comment by idangozlan** [comment]
+1

**Comment by ISDORGA** [comment]
+1

**Comment by SvenC56** [comment]
+1

**Comment by DanPurdy** [comment]
Please stop adding +1s it doesn't help at all

**Comment by EsrefDurna** [comment]
+1

**Comment by adelegard** [comment]
+1

**Comment by davidmpaz** [comment]
Hi,

is there any advance for this issue ? PR is made already. So ETA for merging into upstream ?

cheers

**Comment by cport1** [comment]
Is this still being worked on?

**Comment by zewa666** [comment]
Would really like to see that available as well.

**Comment by ojoven** [comment]
+1

**Comment by iHaiduk** [comment]
+1

**Comment by stephanebruckert** [comment]
New repo just for this https://github.com/srowhani/sass-lint-auto-fix
